pkg/scheduler/scheduler.go

```diff
@@ -319,6 +320,31 @@ func (s *Scheduler) nominate(ctx context.Context, workloads []workload.Info, sna
 	return entries
 }
 
+// resourcesToReserve calculates how much of available resources in cq assignment should be reserved.
+func resourcesToReserve(e *entry, cq *cache.ClusterQueue) cache.FlavorResourceQuantities {
+	if e.assignment.Borrowing || e.assignment.RepresentativeMode() != flavorassigner.Preempt {
+		return e.assignment.Usage
+	}
+	reservedUsage := e.assignment.Usage
+	for flavor, resourceUsage := range e.assignment.Usage {
+		cqQuota := make(map[corev1.ResourceName]*cache.ResourceQuota)
+		for _, rg := range cq.ResourceGroups {
+			for _, cqFlavor := range rg.Flavors {
+				if cqFlavor.Name == flavor {
+					cqQuota = cqFlavor.Resources
+					goto calaculateReservation
```

Comment:
This doesn't look accurate. An assignment could have resources contained in different resource groups, and assigned different flavors.